jax/random.py

```diff
@@ -292,7 +300,7 @@ def _random_bits(key, bit_width, shape):
 
 def _check_shape(name, shape, *param_shapes):
   try:
-    shape = tuple(map(int, shape))
+    shape = tuple(map(lambda x: x.__index__(), shape))
```

Comment:
The standard way to call `__index__` methods is with operator.index:
```suggestion
    shape = tuple(map(operator.index, shape))
```

Reply:
@shoyer Agree. I tried using `operator.index` originally, but `operator.index(Poly({}))` fails with `TypeError: __index__ returned non-int (type Poly)`. `Poly({}).__index__()` runs fine, however. The suggested change would make `test_shapecheck_uniform` fail.

`operator.index` was apparently not designed with anything else than integer indices in mind. But that's exactly what we need here, so I use `__index__` directly as a workaround.

Reply:
Can you leave a comment to this effect?

Reply:
I don't think it makes sense to implement `__index__` on Poly then, since this violates Python's contract for the special method. In theory this could cause trouble if somehow the Poly class leaks into user code. I would just write your own helper function, e.g.,
```
def my_index(x):
   return x if isinstance(x, Poly) else operator.index(x)
```

Reply:
@shoyer Good point. Extracted a custom method now. @mattjj Clarified the choice there.